app/views/admin/versions/_object_desc_and_link.html.haml

```diff
@@ -1,156 +1,162 @@
 
-- elsif version.item_type == 'Subscription' || version.item_type == 'Registration'
+- when 'Subscription', 'Registration'
   = 'conference'
   = link_to Conference.find(version.conference_id).title,
           admin_conference_registrations_path(conference_id: Conference.find(version.conference_id).short_title)
 
-- elsif version.item_type == 'Commercial'
-  - commercial_last_version = get_version_object(PaperTrail::Version.where(item_type: version.item_type, item_id: version.item_id).last)
-  - commercialable_last_version = get_version_object(PaperTrail::Version.where(item_type: commercial_last_version.commercialable_type,
-                                                          item_id: commercial_last_version.commercialable_id).last)
+- when 'Commercial'
+  - commercial = current_or_last_object_state(version)
+  - commercialable_version = PaperTrail::Version.where(item_type: commercial.commercialable_type, item_id: commercial.commercialable_id).last
+  - commercialable = commercialable_version.item || commercialable_version.reify
 
-  - case commercial_last_version.commercialable_type
+  - case commercial.commercialable_type
   - when 'Event'
-    =  link_to 'commercial',
-            edit_admin_conference_program_event_path(conference_id: Conference.find(version.conference_id).short_title,
-                id: commercialable_last_version.id, anchor: 'commercials-content')
-    = "in event #{commercialable_last_version.title}"
+    commercial in event
+    =  link_to commercialable.title,
+           admin_conference_program_event_path(conference_id: Conference.find(version.conference_id).short_title, id: commercialable.id)
 
   - when 'Venue'
-    - if Venue.find_by(id: commercialable_last_version.id)
-      =  link_to 'commercial',
-              edit_admin_conference_program_event_path(conference_id: Conference.find(version.conference_id).short_title,
-                  id: commercialable_last_version.id, anchor: 'commercials-content')
-    - else
-      = 'commercial'
-    = "in venue #{commercialable_last_version.name}"
+    commercial in venue
+    = link_if_alive commercialable_version, commercialable.name,
+                              edit_admin_conference_venue_path(conference_id: Conference.find(version.conference_id).short_title,
+                                              id: commercialable.id, anchor: 'commercials-content')
 
   - when 'Conference'
     = link_to 'commercial',
             admin_conference_commercials_path(conference_id: Conference.find(version.conference_id).short_title)
 
-- elsif %w(EventsRegistration Comment Vote).include?(version.item_type)
+- when 'EventsRegistration', 'Comment', 'Vote', 'Event'
   = 'event'
-  - event_id = get_version_object(version).try(:event_id) || get_version_object(version).try(:commentable_id)
+  - object = current_or_last_object_state(version)
+  - event_id = object.try(:id) || object.try(:event_id) || object.try(:commentable_id)
   = link_to Event.find(event_id).title,
```

Comment:
I've changed this, now it first check if the event exists, if not it will get the name from its last version.